gems/kaminari-core/1.2/kaminari-core.rbs

```diff
@@ -0,0 +1,40 @@
+module Kaminari
+  def self.config: () -> Kaminari::Config
+  def self.configure: () { (Kaminari::Config) -> void } -> void
+
+  class Config
+    attr_accessor default_per_page: Integer
+    attr_accessor max_per_page: Integer?
+    attr_accessor window: Integer
+    attr_accessor outer_window: Integer
+    attr_accessor left: Integer
+    attr_accessor right: Integer
+    attr_accessor page_method_name: Symbol | String
+    attr_accessor max_pages: Integer?
+    attr_accessor params_on_first_page: bool
+
+    attr_writer param_name: Symbol | String | _Caller
+
+    def param_name: () -> (Symbol | String)
+  end
+
+  module PageScopeMethods[Relation] : _Model_Relation[Relation]
+    def per: (Integer num, ?Integer max_per_page) -> Relation
+    def total_pages: () -> Integer
+    def current_page: () -> Integer
+    def next_page: () -> Integer?
+    def prev_page: () -> Integer?
+    def first_page?: () -> bool
+    def last_page?: () -> bool
+    def out_of_range?: () -> bool
+  end
+
+  interface _Model_Relation[Relation]
+    def limit: (Integer) -> Relation
+    def offset: (Integer) -> Relation
+  end
+
+  interface _Caller
```

Comment:
IMO: `_Caller` naming feels a little bit strange. In programming terms, caller means method caller.
How about `_Callable` instead? `-able` is the major pattern for interface naming.